astropy/io/votable/tests/converter_test.py

```diff
@@ -41,9 +40,8 @@ def test_oversize_char():
 
 def test_char_mask():
     config = {'verify': 'exception'}
-    field = tree.Field(
-        None, name='c', datatype='char',
-        config=config)
+    with pytest.warns(exceptions.W47):
+        field = tree.Field(None, name='c', datatype='char', config=config)
```

Comment:
Just a general comment - I think it would be nice to make sure in each case that we know why the warning is being raised. Currently if I look at this test it's not clear why this specific line of code should raise say W47, and we need to make sure that it is correct that the warning is being emitted. Why does it warn and not raise an error given the ``config``? Is that expected?

Reply:
Good point! 😱

Reply:
For this particular line, it is from `W47: ?:?:?: W47: Missing arraysize indicates length 1`:
```
...\io\votable\converters.py in __init__(self, field, config, pos)
    300
    301         if field.arraysize is None:
--> 302             vo_warn(W47, (), config, pos)
    303             field.arraysize = '1'
```

So, this one is okay to ignore. As for the others, sigh, I need to find another day to Sherlock through the warnings...

Reply:
There is no logic to use `config` there and I don't know if that is a bug or feature. 🙈

Reply:
For the deeper design question on how much `config` should control how all the warnings are handled, I have opened a separate issue at #8775